R/whc.R

```diff
@@ -668,9 +671,11 @@ sptf_whc12 <- function(A_CLAY_MI, A_SAND_MI, D_BDS,
                        mp_wp = 1500, mp_fc = 33) {
   # Check input
   arg.length <- max(length(A_CLAY_MI), length(A_SAND_MI), length(D_BDS))
-  checkmate::assert_numeric(A_CLAY_MI, lower = 0, upper = 100, len = arg.length)
-  checkmate::assert_numeric(A_SAND_MI, lower = 0, upper = 100, len = arg.length)
-  checkmate::assert_numeric(D_BDS, lower = 0, upper = 100, len = arg.length)
+  check_numeric('A_CLAY_MI', A_CLAY_MI, FALSE, arg.length)
+  check_numeric('A_SAND_MI', A_SAND_MI, FALSE, arg.length)
+  check_numeric('D_BDS', D_BDS, FALSE, arg.length)
+  checkmate::assert_subset(mp_fc, choices = c(10, 33), any.missing = FALSE)
+  checkmate::assert_numeric(mp_wp, any.missing = FALSE)
```

Comment:
are there no min or max values from literature?

Reply:
Not sure about mp_wp, I'll add it to the todolist. The function description indicates mp_fc must be either 10 or 33.